File: C++/Lab/number_pyramid.cpp
#include <iostream>
using namespace std;

void outputPyramid(int n, int k);

int main(void)
{
    int t;
    int n, k;

    cin >> t;

    for(int i=0; i<t; i++)
    {
        cin >> n >> k;
        outputPyramid(n, k);
    }

    return 0;
}

void outputPyramid(int n, int k)
{
    cout << k;
    int sum = k;
    for (int i=0; i<k-1; i++)
    {
        sum = sum + n - (i + 1);
        cout << "*" << sum;
    }
    cout << endl;
}

File: Java/play_with_number/src/NumberPyramid.java
import java.util.Scanner;

public class NumberPyramid {
	static void outputPyramid(int n, int k) {
		for (int i=0; i<n; i++) {
			if (i + 1 == k) {
				int sum = i + 1;
				System.out.print(sum);
				
				for (int j=0; j<i; j++) {
					sum = sum + n - (j + 1);
					System.out.print("*" + sum);
				}
				
				break;
			}
			else {
				continue;
			}
		}
	}
	
	public static void main(String[] args) {
		Scanner inputReader = new Scanner(System.in);
		System.out.print("How many times to try? > ");
		int t = inputReader.nextInt();
		
		for(int i=0; i<t; i++) {
			System.out.print("How many lines to create? > ");
			int n = inputReader.nextInt();
			System.out.print("Enter a line number to print. > ");
			int k = inputReader.nextInt();
			
			outputPyramid(n, k);
			System.out.println();
		}
	}
}